### DISCUSSION TITLE
the entire cluster space will reduced when disk offline

### DISCUSSION THREAD

**Discussion by jiangfucheng** [minio/minio#13508] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Hi, I have some problem for this PR(https://github.com/minio/minio/pull/11852). When some disk offline, the number of parity blocks will increase, then it will cause a problem:
the space of entire cluster will reduced, because there are fewer data blocks in this erasure set, it will take up more space. so, when this erasure set is full, the entire cluster will can not write any object, it will waste some disk space.

**Describe the solution you'd like**

I found it can re-shard that data in this PR, but i didn't find the relevant code. is it possible add some option to turn off this feature? Or supply some method to re-shard this data when healing. 
Thanks.

**Describe alternatives you've considered**

**Additional context**

**Comment by klauspost** [comment, marked as answer]
@jiangfucheng We don't do resharding. It is expensive and there is little to no gains, so we have no plans for it.

You should not run your cluster in a degraded state long enough for this to matter. 

If it is, download and reupload your objects.

**Reply by jiangfucheng** [reply]
Thanks for your reply. 
1.In our scenario, we transfer large data from old oss to MinIO(more than 1PB each time), each transfer will last for several days. if some disk goes offline during this period, we may not be able to find this problem immediately, then it will cause some disks of that erasure set waste a lot of space(10% used more than other disks). 
2.sometimes we could need stable parity number for our MinIO cluster, because it's controllable. 
> If it is, download and reupload your objects.

for this, how do i check if the parity blocks number of object has been modified, then i need download these objects and reupload.

Thanks.

**Reply by jiangfucheng** [reply]
and then i create this PR(https://github.com/minio/minio/pull/13511) to control this behaivor.
